src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.Small.cs

```diff
@@ -81,8 +82,7 @@ protected override void InitializeConstructorArgumentCaches(ref ReadStack state,
         {
             JsonTypeInfo typeInfo = state.Current.JsonTypeInfo;
 
-            typeInfo.CreateObjectWithArgs ??=
-                JsonSerializerOptions.MemberAccessorStrategy.CreateParameterizedConstructor<T, TArg0, TArg1, TArg2, TArg3>(ConstructorInfo!);
```

Comment:
This was calling into reflection in the deserialization hot path -- replaced with an override to the JsonTypeInfo reflection configuration callback.

Reply:
> This was calling into reflection in the deserialization hot path

For source-gen I believe the property would not have been `null`. Is the reflection now trimmed out/any size savings?

Reply:
> For source-gen I believe the property would not have been null

You're right, it appears we're only using the large constructor converter in source gen.

> Is the reflection now trimmed out/any size savings?

I haven't run any tests, but I figure it should now able to trim all the Ref.Emit components (warnings on that particular `MemberAccessorStrategy` property were being suppressed for some reason). 

cc @eerhardt